src/stores/connectors/ethereum/sign.go

```diff
@@ -227,6 +230,8 @@ func (c Connector) sign(ctx context.Context, addr common.Address, data []byte) (
 		}
 
 		if bytes.Equal(crypto.FromECDSAPub(recoveredPubKey), acc.PublicKey) {
+			// add + 27 to recID
+			appendedSignature[len(appendedSignature)-1] += 27
```

Comment:
what is the reasoning behind V changed value +27? V is calculated based on the ChainID and it should be the Signer struct  who calculates that based on the chainID you can see next:
```
func (s EIP155Signer) SignatureValues(tx *Transaction, sig []byte) (R, S, V *big.Int, err error) {
   R, S, V, err = HomesteadSigner{}.SignatureValues(tx, sig)
   if err != nil {
      return nil, nil, nil, err
   }
   if s.chainId.Sign() != 0 {
      V = big.NewInt(int64(sig[64] + 35))
      V.Add(V, s.chainIdMul)
   }
   return R, S, V, nil
}
```

Reply:
okay, I found the reasoning behind the +27 in this EIP-155 https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md, apparently in case of signing for not specific chain and to prevent replay attacks we need to do V {0,1} + 27.
In any case being that the case I would rather create a specific method to make it more explicit
func eip151(signature []byte) []byte {
	l := len(signature)
	hl := l / 2

	R := new(big.Int).SetBytes(signature[:hl])
	S := new(big.Int).SetBytes(signature[hl:l-1])
    V := new(big.Int).SetBytes(signature[l-1])
    V := V + 27
	return append(append(R.Bytes(), S.Bytes()...), V.Bytes()...)
}